server/src/main/java/org/elasticsearch/cluster/NodeConnectionsService.java

```diff
@@ -209,12 +214,36 @@ public void reconnectToNodes(DiscoveryNodes discoveryNodes, Runnable onCompletio
         });
     }
 
-    private class ConnectionTarget {
+    // exposed for testing
+    protected ConnectionTarget connectionTargetForNode(DiscoveryNode node) {
```

Comment:
I don't think we need to expose the whole `ConnectionTarget` out to tests - we could just allow access to the `DisconnectionHistory` for a node and keep the `ConnectionTarget` class private.